lib/instance_agent/plugins/codedeploy/command_executor.rb

```diff
@@ -192,18 +192,18 @@ def most_recent_deployment_dir(deployment_group)
           return unless File.exist? most_recent_install_file_location
           File.open most_recent_install_file_location do |f|
             return f.read.chomp
-          end 
+          end
         end
 
         private
         def default_app_spec(deployment_spec)
-          default_app_spec_location = File.join(archive_root_dir(deployment_spec), app_spec_path)
-          log(:debug, "Checking for app spec in #{default_app_spec_location}")
-          validate_app_spec_hooks(ApplicationSpecification::ApplicationSpecification.parse(File.read(default_app_spec_location)), deployment_spec.all_possible_lifecycle_events)
+          app_spec_location = app_spec_real_path(deployment_spec)
```

Comment:
I haven't checked the calling code, but make sure the name default_app_spec still makes sense (its variable now).

Reply:
I believe having it as `default_app_spec` doesn't make sense, since we are determining. I renamed it to: `app_spec_location` to signify that.